Crawler/Analyzers/PunctuationAnalyzer.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using Crawler.LexicalAnalyzer;
+using System.Collections.Generic;
+using System.Linq;
+using Crawler.Exceptions;
+
+namespace Crawler.Analyzers
+{
+	public class PunctuationAnalyzer : IPunctuationAnalyzer
+	{
+		public double CalculateAverageWordsCountBetweenPunctuation(List<Token> tokens)
+		{
+			if(!tokens.Any()) return 0;
+
+			var wordsCountsBetweenPunctuations = WordsCountsBetweenPunctuations(tokens);
+
+			return !wordsCountsBetweenPunctuations.Any() ? 0 : wordsCountsBetweenPunctuations.Average();
+		}
+
+		private List<int> WordsCountsBetweenPunctuations(List<Token> tokens)
```

Comment:
why the private method is here and not and the bottom of the page? after all the public methods that uses it